### ISSUE TITLE
Tags

### ISSUE THREAD

**Issue by adriatic** [aurelia-ui-toolkits/catalog-admin#3]
### Tags

This project was created with the intent to provide meaningful search for the **`bridge catalog admin`** application, by ___tagging___ catalog's samples  subsequently referred as **`Components`** and variants of each sample referred to  **`Samples`**. See more in the  **[catalog's data structures](https://github.com/aurelia-ui-toolkits/bridge-catalog-admin/issues/4)** article.

The **`bridge catalog admin`** application will support two types of tags:

#### 1. Component tag
```
namespace Catalog.Models
{
    public class ComponentTag
    {
        enum Categories
        {
            Intro, Featured, Reference, Landing, Guide, Example
        };

        Dictionary<Categories, string> tag = new Dictionary<Categories, string>();

    }
}
```

**Note:** this is the categorization used by **[Mozzilla](https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Tag)**. Suggestions for changes / additions **welcome**.

#### 2. Sample tag
```
using System.Collections.Generic;

namespace Catalog.Models
{
    public class SampleTag
    {
        Dictionary<string, string> tag = new Dictionary<string, string>();
    }
}
```

**Note:** I used the generic structure **`Dictionary<string, string>`** since I did not spend enough time thing what should be our **`enum`** that fits the task of editing gists.

***

**Comment by adriatic** [comment]
### @Thanood writes

So would the key be the tag or sample id? The former would provide a slightly faster search I guess. While the latter might be easier to manage.

Also if you have tags, what do you need categories for?

**Comment by adriatic** [comment]
Good comments, although the first one confused me, so I will address the second question first, hoping that the answer to the first one will come on its own, while I am addressing the second:

My desire is to "tag the shit out" of everything we have - including all written online documents, where most of the **[component tags](https://github.com/aurelia-ui-toolkits/bridge-catalog-admin/issues/5#issue-204182000)** will be used for classification and subsequent retrieval. I will also admit that I was thinking about the "supercat" creating this **`enum`** set. 

To address both of your questions at once: I believe that using two level classification will result with better user experience. As an example consider this:

- all KendoUI widgets (Aurelia components) like **`autocomplete`** or **`grid`** will be tagged with "Example" enum from the **`Component`** set and will be used as the primary tag. Everything else related to the chosen Component will be described with multiple tags from the SampleTag collection - and this is where I will appreciate your suggestion for specific words / set of words that could result with a good fetch from user's search.

- all online documents (all of which live in GitHub as markdown files) would be tagged with one of `Intro, Featured, Reference, Landing, Guide` (note that the initial collection is just a start). With this in mind the code samples primary tag will be 'Example", and that will be allow us to differentiate between samples and documents.

Besides obvious questions where the user asks why his app does not work and ends with the creation of the gist in the context of which the explanation comes out with relative ease, I would like to really improve the user self service part of our support, where one could ask "show me the components that use remote data", or any other hopefully more complex search query.

**Comment by Thanood** [comment]
Let me try an example of one of the `autocomplete` samples ([server filtering](http://aurelia-ui-toolkits.github.io/demo-kendo/#/samples/autocomplete-server-filtering)). I will use JSON here to avoid some typing "overhead":

```json
{
  "name": "autocomplete server filtering",
  "category": "example",
  "tags": [ "remote", "input", "editors" ]
}
```

I can see that f.i. users can narrow their search to the `example` category while still being able to use an `OR` type search within the tags itself. If `example` was just another tag, users looking for remote examples would have to search `example && remote`. Using only `remote` they would find documents, too.
Now I can search for `remote || editors` and still keep the `example` restriction.

In short: I agree, I just seem to need an example to get what you're aiming at. 😃
